tensorboard/webapp/persistent_settings/_data_source/persistent_settings_data_source.ts

```diff
@@ -210,14 +210,18 @@ export class OSSSettingsConverter extends SettingsConverter<
 
     if (
       backendSettings.hasOwnProperty('singleSelectionHeaders') &&
-      typeof backendSettings.singleSelectionHeaders === 'object'
+      Array.isArray(backendSettings.singleSelectionHeaders) &&
+      // If the settings stored in the backend are invalid, reset back to default.
+      backendSettings.singleSelectionHeaders[0].name !== undefined
     ) {
```

Comment:
I don't think you need this initial condition anymore. FYI `Array.isArray(undefined)` returns `false`.
```suggestion
    // If the settings stored in the backend are invalid, reset back to default.
    if (Array.isArray(backendSettings).singleSelectionHeaders && backendSettings.singleSelectionHeaders[0].name !== undefined) {
```